### PR TITLE
Integ test beta

### PR DESCRIPTION
**Pull request by nathalapooja**
*Description of changes:*
Implemented the Integration testing for EKS addon on Beta cluster

Successful test run https://github.com/aws/amazon-cloudwatch-agent-operator/actions/runs/6201567912

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sky333999** [comment]
I had assumed all the CFN stuff was temporary to validate that it works before we align it with our existing terraform setup :)

Lets sync up offline - Id like to better understand why the beta and non-beta tests need to be so diff. Ideally, id expect we can reuse most of the cost with a flag to add some extra beta components.

**Comment by sky333999** [comment]
Also some additional enhancements which can be a follow up:
* Would be good to have a way to trigger both beta and non-beta independently and via the GHAs UI.
* We need a way of taking in a version as an option arg that can be supplied via the UI.
* As part of the matrix to run multiple EKS versions (comment from last PR), we should also add ability to test diff addon versions.

**Comment by nathalapooja** [comment]
> Also some additional enhancements which can be a follow up:
> 
> * Would be good to have a way to trigger both beta and non-beta independently and via the GHAs UI.
> * We need a way of taking in a version as an option arg that can be supplied via the UI.
> * As part of the matrix to run multiple EKS versions (comment from last PR), we should also add ability to test diff addon versions.

I believe we should better create a sim to track all these additional enhancements and track it.

**Comment by nathalapooja** [comment]
https://github.com/aws/amazon-cloudwatch-agent-operator/actions/runs/6223907660 successful integration test after the terraform beta changes.

File: integration-tests/eks-addon/validateResources_test.go [hunk 1]
```diff
@@ -18,89 +18,86 @@ import (
 	"testing"
 )
 
-const NAMESPACE = "amazon-cloudwatch"
+const nameSpace = "amazon-cloudwatch"
```
Comment (sky333999) [inline integration-tests/eks-addon/validateResources_test.go:21]: nit: nameSpace -> namespace

**Review by sky333999 — APPROVED** [review]

**Review by lisguo — APPROVED** [review]

**Pull request merged**